### PR TITLE
Stricter WCIF schema checking upon `PATCH`

### PR DESCRIPTION
**Pull request by gregorbg**
This PR does three things:
1. Remove the forced check for `formatVersion` (sigh...) and fall back to `STABLE_VERSION` (at the time of this PR: version `1.1`) if not specified
2. Make the schema check on WCIF stricter by not allowing any `additionalProperties` at all
3. Add tests for the corner cases that previously struck us down

If you look at the diff, use "Hide Whitespace" to better understand the changes (with fat airquotes) that I'm doing to the JSON schema. It's actually redundant to check a separate `type`-only object if the nested `oneOf` already implies the `type` fields as `const`.

### CODE REVIEW

File: app/models/competition.rb [hunk 1]
```diff
@@ -1942,11 +1942,15 @@ def schedule_wcif
     }
   end
 
+  def self.validate_wcif_schema!(wcif, version: WCIF_STABLE_VERSION)
+    expected_schema = Competition.wcif_json_schema(version: version)
+    JSON::Validator.validate!(expected_schema, wcif, noAdditionalProperties: true)
```
Comment (gregorbg) [inline app/models/competition.rb:1947]: This `noAdditionalProperties` thing is the "magic change" at the heart of this PR.

Previously, our check did not care if you sent extra stuff. It only checked if _the properties that we defined in our schema_ were fine, and let the rest be. So if you sent V2 with `participationCondition`, our code was like "I have no schema for that so it's fine". Now it will become "I have no schema for that, in other words its an additonal property, so I complain".

**Review by dunkOnIT — APPROVED** [review]
LGTM!
